lib/facter/pulp_consumer_id.rb

```diff
@@ -0,0 +1,14 @@
+Facter.add(:pulp_consumer_id) do
+  confine kernel: 'Linux'
+  setcode do
+    pulp_consumer_id = nil
+    status = Facter::Util::Resolution.exec('pulp-consumer status')
+    unless status.nil?
+      # Strip color from command output
+      status.gsub!(/\e\[([;\d]+)?m/, '')
+      captures = /^This consumer is registered to the server\s\[(.*)\]\swith\sthe\sID\s\[(.*)\]\.$/.match(status)
```

Comment:
Could do; perhaps it was a bit lazy of me not to. (Personally, I'm likely to only use the fact as if it were a boolean).
I just need to think about the correct way and style.
What is the preferred style?  I *could* create a structured fact??   If not, should each fact be in it's own file?  Does code shared between facts belong in it's own namespace? That sort of thing.  I'll figure it out.  Pretty sure I shouldn't just nest `Facter.add` blocks anyway!